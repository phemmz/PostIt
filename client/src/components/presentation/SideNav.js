import React, { Component } from 'react';

class SideNav extends Component {
	render() {
		return (
      <ul id="slide-out" className="side-nav groupdetails-nav" >
        <li>
          <div className="user-view">
            <div className="background">
              <img className="groupdetails-img" src="https://images.unsplash.com/photo-1493219686142-5a8641badc78?dpr=2&auto=format&fit=crop&w=1500&h=1000&q=80&cs=tinysrgb&crop="/>
            </div>
            <div>
              <a href="#!user"><img className="circle" src=""/></a>
              <a href="#!name"><span className="white-text name">{this.props.currentUser.username}</span></a>
              <a href="#!email"><span className="white-text email right">{this.props.currentUser.email}</span></a>
            </div>
          </div>
        </li>
        <li><a href="#!" className="waves-effect waves-light green nav-anchor"><i className="material-icons">cloud</i>Group Members</a></li>
        <li><div className="divider"></div></li>
        <li><a href="#addUser" className="waves-effect waves-light green nav-anchor modal-trigger"><i className="material-icons">people_outline</i>Add User</a></li>
      </ul>
		)
	}
}
export default SideNav;