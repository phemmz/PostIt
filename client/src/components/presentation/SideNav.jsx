import React from 'react';
import PropTypes from 'prop-types';

const SideNav = ({
  currentUser
}) => {
  return (
    <ul id="slide-out" className="side-nav groupdetails-nav" >
      <li>
        <h5
          className="white-text name user-size"
        >{currentUser.username}</h5>
      </li>
      <li>
        <h5
          className="white-text email user-size"
        >{currentUser.email}</h5>
      </li>
      <li>
        <a
          href="#!"
          className="waves-effect waves-light green nav-anchor"
        ><i className="fa fa-users snav" aria-hidden="true" />Members</a></li>
      <li><div className="divider" /></li>
      <li>
        <a
          href="#addUser"
          className="waves-effect waves-light green nav-anchor modal-trigger"
        ><i className="fa fa-user-plus snav" aria-hidden="true" />Add User</a></li>
    </ul>
  );
};

SideNav.propTypes = {
  currentUser: PropTypes.shape({
    username: PropTypes.string,
    email: PropTypes.string
  }).isRequired
};

export default SideNav;
