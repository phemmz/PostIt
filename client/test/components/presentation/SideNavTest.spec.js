import React from 'react';
import { mount } from 'enzyme';
import SideNav from '../../../src/components/presentation/SideNav';

describe('SideNav Component', () => {
  const currentUser = {
    username: 'phemmz',
    email: 'phemmzmcllroy@gmail.com'
  };
  it('should render a sidenav component', () => {
    const wrapper = mount(<SideNav currentUser={currentUser} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('currentUser')).toBe(currentUser);
    expect(wrapper.find('ul').exists()).toBe(true);
    expect(wrapper.find('li').exists()).toBe(true);
    expect(wrapper.find('h5').exists()).toBe(true);
    expect(wrapper.find('a').exists()).toBe(true);
    expect(wrapper.find('i').exists()).toBe(true);
    expect(wrapper.find('ul').length).toBe(1);
    expect(wrapper.find('li').length).toBe(5);
    expect(wrapper.find('h5').length).toBe(2);
    expect(wrapper.find('a').length).toBe(2);
    expect(wrapper.find('i').length).toBe(2);
  });
});
