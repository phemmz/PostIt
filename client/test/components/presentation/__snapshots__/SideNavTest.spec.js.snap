// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav Component should render a sidenav component 1`] = `
<SideNav
  currentUser={
    Object {
      "email": "phemmzmcllroy@gmail.com",
      "username": "phemmz",
    }
  }
>
  <ul
    className="side-nav groupdetails-nav"
    id="slide-out"
  >
    <li>
      <h5
        className="white-text name user-size"
      >
        phemmz
      </h5>
    </li>
    <li>
      <h5
        className="white-text email user-size"
      >
        phemmzmcllroy@gmail.com
      </h5>
    </li>
    <li>
      <a
        className="waves-effect waves-light green nav-anchor"
        href="#!"
      >
        <i
          aria-hidden="true"
          className="fa fa-users snav"
        />
        Members
      </a>
    </li>
    <li>
      <div
        className="divider"
      />
    </li>
    <li>
      <a
        className="waves-effect waves-light green nav-anchor modal-trigger"
        href="#addUser"
      >
        <i
          aria-hidden="true"
          className="fa fa-user-plus snav"
        />
        Add User
      </a>
    </li>
  </ul>
</SideNav>
`;
